src/frontend/sharedComponents/Editor/index.tsx

```diff
@@ -131,7 +134,45 @@ export const Editor = ({
                     <React.Suspense
                       key={att.driveDiscoveryId + att.hash + att.type}
                       fallback={<ThumbnailLoader size={size} />}>
-                      <SavedPhotoThumbnailImage size={size} photo={att} />
+                      <SavedPhotoThumbnailImage
+                        size={size}
+                        photo={att}
+                        onPress={() => {
+                          if (!observationId) {
+                            navigate('PhotoPreviewModal', {
+                              photo: att,
+                            });
+                            return;
+                          }
+
+                          // TODO: Ideally use the core-react hooks but the error handling makes it tricky
```

Comment:
didn't want to go down a rabbithole of making changes to something like the Editor component to make this more elegant. in the context of what this is for, I think this is generally okay for now.

could be adjusted if we decide to make the PhotoPreviewModal fetch the `createdByDeviceId` value internally as opposed to supplying it via a nav param. Wasn't too keen on doing that as an initial implementation, but open to considering it if necessary.

Reply:
the top level `catch` should also handle catches from the internal promise (note the returning of the internal promise i.e. `return projectApi.$originalVersionIdToDeviceId(...`).

whether this is the best way is debatable! like the comment mentions, i think it's partially a symptom of how this component is set up, but maybe moving the fetching to the photo preview screen is what we should do anyways.

Reply:
ended up moving the loading of this info into the screen. see https://github.com/digidem/comapeo-mobile/pull/1179/commits/f7961cf39c5b0eafb3b4c2d5457efa244241d0c4